docs/adr/0016-use-graphql-and-apollo-for-settings-redesign.md

```diff
@@ -0,0 +1,138 @@
+- The React Settings application will only have to manage interacting with a single endpoint and Apollo Client makes managing the data in our React app straightforward.
+- Does not preclude option C, as we can replace or supplement direct calls to the FxA auth-server down the line. This is faster for initial development, and this option over option C will also help mitigate risk around using a relatively novel piece of technology for FxA.
+
+## Pros and Cons of the Options
+
+### Option A - Send standard CRUD requests to our REST endpoints
+
+#### Description
+
+- Don't use GraphQL. Send all requests to their corresponding endpoints.
+
+##### Pros:
+
+- No novel tech.
+- No additional setup time.
+- No new technical debt.
+- Our REST API setup has been vetted by years of use in production.
+
+##### Cons:
+
+- We'd still want to work towards optimizing the way we do our Settings page network calls, which has around 17 executed sequentially. REST results in either overfetching data, doing more queries than is needed, or spending additional time manually optimizing calls as we update the Settings page.
+- We may need to figure out a more ideal way to organize our API calls than using `fxa-payments-server` as a blueprint.
+- We'll be turning down all of the benefits laid out in the GraphQL section, like clarity on the client-side around what data is being sent and received.
+- If we later choose to use GraphQL, even with the option B approach, we'll face a lot of refactoring at a later time that could be mitigated now.
+
+### Options B & C - use GraphQL and Apollo
+
+#### Description
+
+This section outlines the pros and cons for using GraphQL and Apollo in general and applies to both options B and C.
+
+While GraphQL will be novel tech to the FxA stack, it's not entirely novel to the FxA ecosystem. That is to say, the Admin Panel project (`fxa-admin-panel`/`fxa-admin-server` packages) served as a prototype for experimenting with GQL and Apollo.
+
+##### Pros
+
+- We can use GQL for all Settings Redesign needs and won't need to mix GQL and REST.
+- GQL can consolidate network requests. Even if some GQL queries need to be fed into the auth-server (option B), the client will be relieved of a heavy network request burden and instead, a single request from the client asks the server to perform all of the requests. This will offer a performance boost especially on the main Settings page, where currently around 17 different requests must be made.
+- Since only required data will be requested, the data returned won't be overfetched like it is in traditional REST architecture, resulting in a smaller payload.
+- GQL offers a flexible and declarative approach to data - both FE and BE developers can focus on describing the data rather than implementing and optimizing numerous REST endpoints. This makes it painfully clear what's being sent and received between client and server which can improve visibility into what's happening in components.
```

Comment:
Ah, I think this is the point I was looking for in earlier comments. One of the key GraphQL vs REST points, IMO, is moving data requirement schemas from server closer to client where the data is consumed. REST patterns tend to define resources on the server, GraphQL accepts data requirements from the client.

Reply:
I expounded upon this a bit from your comment (pushing changes up soon).